sci-physics/lhapdf/lhapdf-6.3.0-r2.ebuild

```diff
@@ -78,6 +79,7 @@ src_install() {
 
 	cd "${S}"/wrappers/python || die
 	distutils-r1_src_install
+	rm -rf "${D}/$(python_get_sitedir)"/*.egg-info || die
```

Comment:
`-f` negates `|| die` so we shouldn't use it.

I'm also not sure if removal of these files is the correct solution. They should not be present since we have `DISTUTILS_USE_PEP517` here, perhaps something else in the build system is still calling legacy `setup.py install`.

Reply:
I took it from this commit https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=7d23e75b2b7242c31b33bdc7b869edf599002262

Reply:
The setup.py call is here https://github.com/gentoo/gentoo/blob/master/sci-physics/lhapdf/files/lhapdf-6.3.0-egg.patch and it is actually just distutils.core.setup() (https://gitlab.com/hepcedar/lhapdf/-/blob/lhapdf-6.3.0/wrappers/python/setup.py.in?ref_type=tags)

Reply:
> I took it from this commit https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=7d23e75b2b7242c31b33bdc7b869edf599002262

That change was undone in https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=5a37c82fd3188758ae0b68f7347176fbdcf099a9

Reply:
The short summary of this QA warning is that `.egg-info` is the old setup.py thing, and `.dist-info` is the new PEP517 thing. If we remove the egg-info without adding a dist-info, we lose the module installation information at runtime.

Reply:
I suspect that `--enable-python` in the configure phase is responsible for installing the `.egg-info`. Perhaps we can simply disable this, given that the ebuild already calls the `distutils-r1*` functions, those functions should install the module "properly" (i.e. with the `.dist-info`). Alternatively, if adjusting the configure breaks things, then we can remove the `.egg-info`, if (and only if) the package also installs a `.dist-info`.

Reply:
But there are a few packages that remove it this way: https://github.com/search?q=repo%3Agentoo%2Fgentoo+rm+egg-info&type=code (even one in the distutils.eclass). The python module works without the egg files, since there are dist-info files.

```
$ equery files lhapdf
 * Searching for lhapdf ...
  * Contents of sci-physics/lhapdf-6.3.0-r2:
  /usr
  /usr/bin
  /usr/bin/lhapdf
  /usr/bin/lhapdf-config
  /usr/include
  /usr/include/LHAPDF
  /usr/include/LHAPDF/AlphaS.h
  /usr/include/LHAPDF/BicubicInterpolator.h
  /usr/include/LHAPDF/BilinearInterpolator.h
  /usr/include/LHAPDF/Config.h
  /usr/include/LHAPDF/ContinuationExtrapolator.h
  /usr/include/LHAPDF/ErrExtrapolator.h
  /usr/include/LHAPDF/Exceptions.h
  /usr/include/LHAPDF/Extrapolator.h
  /usr/include/LHAPDF/Factories.h
  /usr/include/LHAPDF/FileIO.h
  /usr/include/LHAPDF/GridPDF.h
  /usr/include/LHAPDF/Info.h
  /usr/include/LHAPDF/Interpolator.h
  /usr/include/LHAPDF/KnotArray.h
  /usr/include/LHAPDF/LHAGlue.h
  /usr/include/LHAPDF/LHAPDF.h
  /usr/include/LHAPDF/LogBicubicInterpolator.h
  /usr/include/LHAPDF/LogBilinearInterpolator.h
  /usr/include/LHAPDF/NearestPointExtrapolator.h
  /usr/include/LHAPDF/PDF.h
  /usr/include/LHAPDF/PDFIndex.h
  /usr/include/LHAPDF/PDFInfo.h
  /usr/include/LHAPDF/PDFSet.h
  /usr/include/LHAPDF/Paths.h
  /usr/include/LHAPDF/Reweighting.h
  /usr/include/LHAPDF/Utils.h
  /usr/include/LHAPDF/Version.h
  /usr/lib
  /usr/lib/python3.11
  /usr/lib/python3.11/site-packages
  /usr/lib/python3.11/site-packages/LHAPDF-6.3.0.dist-info
  /usr/lib/python3.11/site-packages/LHAPDF-6.3.0.dist-info/METADATA
  /usr/lib/python3.11/site-packages/LHAPDF-6.3.0.dist-info/WHEEL
  /usr/lib/python3.11/site-packages/LHAPDF-6.3.0.dist-info/top_level.txt
  /usr/lib/python3.11/site-packages/lhapdf.cpython-311-x86_64-linux-gnu.so
  /usr/lib64
  /usr/lib64/libLHAPDF.so
  /usr/lib64/pkgconfig
  /usr/lib64/pkgconfig/lhapdf.pc
  /usr/share
  /usr/share/LHAPDF
  /usr/share/LHAPDF/lhapdf.conf
  /usr/share/LHAPDF/pdfsets.index
  /usr/share/doc
  /usr/share/doc/lhapdf-6.3.0-r2
  /usr/share/doc/lhapdf-6.3.0-r2/AUTHORS.bz2
  /usr/share/doc/lhapdf-6.3.0-r2/ChangeLog.bz2
  /usr/share/doc/lhapdf-6.3.0-r2/README.bz2
  /usr/share/doc/lhapdf-6.3.0-r2/TODO.bz2
```

Reply:
> The python module works without the egg files, since there are dist-info files.

Great! Then it should be safe to remove them. We maybe still want to remove `--enable-python` though, since we compile and install the python module via the eclass anyway, doing so via the main build system probably results in doing it twice. And the build system usually does not respect the python targets set for the ebuild. To test if it does, you can intentionally mismatch your `/etc/python-exec/python-exec.conf` with your `PYTHON_SINGLE_TARGET`.

Reply:
AFAIK, `--enable-python` is needed for cython to run to generate one of the files to be compiled against the right version of python.

Reply:
> AFAIK, `--enable-python` is needed for cython to run to generate (at least) one of the files to be compiled against the right version of python.

Alright, then lets keep it around.

Reply:
targets `python3_{11,12}` work individually and only with that version.